Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -4403,8 +4403,8 @@ void Document::update_for_history_step_application(JS::NonnullGCPtr<HTML::Sessio
         //    - previousEntryForActivation's document state's origin is same origin with document's origin; and
         //    - previousEntryForActivation's document's initial about:blank is false,
         if (navigation_type.value() == Bindings::NavigationType::Replace
-            && previous_entry_for_activation.value()->document_state()->origin() == origin()
-            && !previous_entry_for_activation.value()->document()->is_initial_about_blank()) {
+            && previous_entry_for_activation->document_state()->origin() == origin()
+            && !previous_entry_for_activation->document()->is_initial_about_blank()) {
```

Comment:
The crash I mention is on this line due to a null document - still trying to figure out how this is meant to be set, or why it is getting set to null